python/ray/data/datasource/partitioning.py

```diff
@@ -225,28 +225,28 @@ class PathPartitionParser:
     Path-based partition formats embed all partition keys and values directly in
```

Comment:
i think if we remove `PathPartitionEncoder` pending discussion above, we may also just be able to remove `PathPartitionParser` as well

Reply:
I kept `DeveloperAPI` for `PathPartitionParser`, because this class is referenced in `Datasource`, which is a `PublicAPI`. Therefore, the lint check expects this class to have an API stability annotation.